docs/src/main/sphinx/admin/fault-tolerant-execution.rst

```diff
@@ -0,0 +1,338 @@
+
+       May be overridden for the current session with the
+       ``fault_tolerant_execution_max_task_split_count``
+       :ref:`session property <session-properties-definition>`.
+     - ``256``
+
+Node allocation
+^^^^^^^^^^^^^^^
+
+With a task retry policy, nodes are allocated to tasks based on available memory
+and estimated memory usage. If task failure occurs due to exceeding available
+memory on a node, the task is restarted with a request to allocate the full node
+for its execution.
+
+The initial task memory-requirements estimation is static and configured with
+the ``fault-tolerant-task-memory`` configuration property. This property only
+applies to a ``TASK`` retry policy.
+
+.. list-table:: Node allocation configuration properties
+   :widths: 30, 40, 30
+   :header-rows: 1
+
+   * - Property name
+     - Description
+     - Default value
+   * - ``fault-tolerant-execution-task-memory``
+     - Initial task memory estimation used for bin-packing when allocating nodes
+       for tasks. May be overridden for the current session with the
+       ``fault_tolerant_execution_task_memory``
+       :ref:`session property <session-properties-definition>`.
+     - ``4GB``
+
+Other tuning
+^^^^^^^^^^^^
+
+The following additional configuration properties can be used to manage
+fault-tolerant execution:
+
+.. list-table:: Other fault-tolerant execution configuration properties
+   :widths: 30, 40, 30, 30
+   :header-rows: 1
+
+   * - Property name
+     - Description
+     - Default value
+     - Retry policy
+   * - ``fault-tolerant-execution-task-descriptor-storage-max-memory``
+     - Maximum amount of memory to be used to store task descriptors for fault
+       tolerant queries on coordinator. Extra memory is needed to be able to
+       reschedule tasks in case of a failure.
+     - (JVM heap size * 0.15)
+     - Only ``TASK``
+
+.. _fte-exchange-manager:
+
+Exchange manager
+----------------
+
+Exchange spooling is responsible for storaging and managing spooled data for
+fault-tolerant execution. By default, spooled data is stored in the coordinator
+node's memory and is subject to memory resource limits. You can configure a
+filesystem-based exchange manager that stores spooled data in a specified
+location, either a local filesystem or an S3-compatible storage server.
+
+To configure an exchange manager, create a new ``exchange-manager.properties``
+configuration file on the coordinator, such as
+``etc/exchange-manager.properties``. In this file, set the
+``exchange-manager.name`` configuration propertry to ``filesystem``, and
+additional configuration properties as needed for your storage solution.
+
+.. list-table:: Exchange manager configuration properties
+   :widths: 30, 40, 30
+   :header-rows: 1
+
+   * - Property name
+     - Description
+     - Default value
+   * - ``exchange.base-directory``
+     - The base directory URI location that the exchange manager uses to store
+       spooling data.
```

Comment:
We should probalby say here (or somewhere else) that currently only `s3*://` and local filesystems are supported